sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer_client.py

```diff
@@ -193,31 +156,37 @@ def receive(self, on_events, consumer_group, **kwargs):
                 :caption: Receive events from the EventHub.
         """
         partition_id = kwargs.get("partition_id")
-
         with self._lock:
             error = None
-            if (consumer_group, '-1') in self._event_processors:
-                error = ValueError("This consumer client is already receiving events from all partitions for"
-                                   " consumer group {}. ".format(consumer_group))
+            if (consumer_group, ALL_PARTITIONS) in self._event_processors:
+                error = ("This consumer client is already receiving events "
+                         "from all partitions for consumer group {}.".format(consumer_group))
             elif partition_id is None and any(x[0] == consumer_group for x in self._event_processors):
-                error = ValueError("This consumer client is already receiving events for consumer group {}. "
-                                   .format(consumer_group))
+                error = ("This consumer client is already receiving events "
+                         "for consumer group {}.".format(consumer_group))
             elif (consumer_group, partition_id) in self._event_processors:
-                error = ValueError("This consumer is already receiving events from partition {} for consumer group {}. "
-                                   .format(partition_id, consumer_group))
+                error = ("This consumer client is already receiving events "
+                         "from partition {} for consumer group {}. ".format(partition_id, consumer_group))
             if error:
                 log.warning(error)
-                raise error
+                raise ValueError(error)
 
             event_processor = EventProcessor(
-                self, consumer_group, on_events,
+                self, consumer_group, on_event,
                 partition_manager=self._partition_manager,
                 polling_interval=self._load_balancing_interval,
                 **kwargs
             )
-            self._event_processors[(consumer_group, partition_id or "-1")] = event_processor
-
-        event_processor.start()
+            self._event_processors[(consumer_group, partition_id or ALL_PARTITIONS)] = event_processor
+        try:
+            event_processor.start()
+        finally:
+            event_processor.stop()
+            with self._lock:
+                try:
+                    del self._event_processors[(consumer_group, partition_id or ALL_PARTITIONS)]
+                except KeyError:
```

Comment:
Is there an actual way we get corrupted to the level we're not even sure the dict we own contains the right EventProcessor?

Reply:
If a process were to call ConsumerClient.close(), that would clear the internal dict:
https://github.com/yunhaoling/azure-sdk-for-python/blob/feature/eventhubs_refactor/sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer_client.py#L210

So it's possible for this processor to be shut down be an external process.